tests/storage/lvm_test.py

```diff
@@ -137,13 +139,14 @@ def test_build_command_long_filter(fake_devices):
 def test_rebuild_filter_after_invaliation(fake_devices):
     # Check that adding a device and invalidating the filter rebuilds the
     # config with the correct filter.
-    lc = lvm.LVMCache()
-    lc._addExtraCfg(["lvs"])
-
+    fake_runner = FakeRunner()
+    lc = lvm.LVMCache(fake_runner)
     fake_devices.append("/dev/mapper/c")
     lc.invalidateFilter()
 
-    cmd = lc._addExtraCfg(["lvs"])
+    lc.run_command(["lvs"])
+    cmd = fake_runner.calls[0]
+
     assert cmd[3] == build_config(fake_devices)
```

Comment:
Change makes sense, testing public functions is better. But this change should be done first before we make any other change to make sure it is correct with the current code and we don't break the code while adding the new feature.